File: Data Structures & Algorithms/majority-element/submission-3.java
class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer,Integer> mp=new HashMap<>();

        for(int num:nums){
            mp.put(num,mp.getOrDefault(num,0)+1);
        }

        int ans=0;

        for(Map.Entry<Integer,Integer> entry:mp.entrySet()){
            int key=entry.getKey();
            int val=entry.getValue();
            if(val>nums.length/2){
                ans=key;
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/intersection-of-two-arrays/submission-3.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        if not nums1 or not nums2:
            return []

        st=set()
        
    

        for val in nums2:
            if val in nums1:
                st.add(val)
    
        return list(st)




        return list(ans)